pkg/sideinputs/store/jetstream/watcher.go

```diff
@@ -0,0 +1,194 @@
+package jetstream
+
+import (
+	"context"
+	"fmt"
+	"github.com/nats-io/nats.go"
+	jsclient "github.com/numaproj/numaflow/pkg/shared/clients/nats"
+	"github.com/numaproj/numaflow/pkg/shared/logging"
+	"github.com/numaproj/numaflow/pkg/sideinputs/store"
+	"go.uber.org/zap"
+	"time"
+)
+
+// jetStreamWatch implements the watermark's KV store backed up by Jetstream.
+type jetStreamWatch struct {
+	pipelineName      string
+	kvBucketName      string
```

Comment:
let's consider writing a generic watcher which can be used for both watermark and side input?  should be relatively easy with generic interfaces.